lib/commands/xctest.js

```diff
@@ -0,0 +1,87 @@
+import B from 'bluebird';
+import { logger } from 'appium-support';
+import log from '../logger';
+import appManagement from './app-management';
+
+let helpers = {}, commands = {}, extensions = {};
+
+const xctestLog = logger.getLogger('XCTest');
+
+function assertIDB (context) {
+  if (!context.opts?.device?.idb) {
+    log.errorAndThrow(`To use XCTest runner, IDB (https://github.com/facebook/idb) must be installed ` +
+      `and sessions must be run with the "launchWithIDB" capability`);
+  }
+}
+
+/**
+ * Run an XCTest
+ *
+ * @param {object} params - The parameters that get passed to IDB to run XC test
+ *                          * testRunnerBundleId: Bundle ID of the test app
+ *                          * appUnderTestBundleId: Bundle ID of the app-under-test
+ *                          * xcTestBundleId: Bundle ID of the testrunner
+ *                          * testType: 'app', 'ui', or 'logic'
+ *                          * opts: Object containing opts.env (environment variables) and
+ *                              opts.args (test arguments)
+ */
+commands.mobileRunXCTest = async function runXCTest ({
+  testRunnerBundleId,
+  appUnderTestBundleId,
+  xctestBundleId,
+  testType = 'ui',
+  opts = {},
+}) {
+  assertIDB(this);
+  const subproc = await this.opts.device.idb.runXCUITest(
+        testRunnerBundleId, appUnderTestBundleId, xctestBundleId, {...opts, testType},
+  );
+
+  const p = new B((resolve, reject) => {
+    subproc.on('output', (stdout, stderr) => {
+      stdout && xctestLog.info(stdout);
+      stderr && xctestLog.error(stderr);
+    });
+
+    subproc.on('exit', (code, signal) => {
+      if (code > 0) {
+        return reject({code, signal, message: `Could not run XCTest. Exit status '${code}'`});
+      }
+      resolve(null);
+    });
+  });
+  return await p;
+};
+
+/**
+ * Install an XCTestBundle
+ *
+ * @param {string} xctestBundleId - Bundle ID of the test app
+ */
+commands.mobileInstallXCTestBundle = async function installXCTestBundle (xctestBundle) {
+  assertIDB(this);
+  xctestLog.info(`Installing bundle '${xctestBundle}'`);
+  return await appManagement.mobileInstallApp.call(this, {app: xctestBundle});
```

Comment:
I assume we don't need to call `mobileInstallApp` here. It would be enough to call `await this.opts.device.installApp` instead

Reply:
also, does this work in general? I thought xctest bundles could only be installed via idb

Reply:
I changed it to use the xctest installer.